WcaOnRails/spec/controllers/api_competitions_controller_spec.rb

```diff
@@ -40,4 +40,234 @@
       expect(parsed_body["website"]).to eq "http://example.com"
     end
   end
+
+  describe 'GET #index' do
+    it 'sorts newest to oldest' do
+      yesteryear_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.year.ago)
+      yesterday_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.ago)
+      today_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 0.days.ago)
+      tomorrow_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.from_now)
+
+      get :index
+      expect(response.status).to eq 200
+      json = JSON.parse(response.body)
+      expect(json.map { |c| c["id"] }).to eq [tomorrow_comp, today_comp, yesterday_comp, yesteryear_comp].map(&:id)
```

Comment:
There isn't actually a documented standard for this particular API (note this is particular api is not serving up WCIF, this is a "give me a list of competitions that match some parameters" api). I think that including the full string here is going to be kind of crazy. Here's what the full thing looks like:

<details>

```
[
  {
    "class": "competition",
    "url": "/competitions/FooComp62015",
    "id": "FooComp62015",
    "name": "Foo Comp 6 2015",
    "website": "https://www.worldcubeassociation.org",
    "short_name": "Foo Comp 6 2015",
    "city": "San Francisco",
    "country_iso2": "US",
    "start_date": "2017-04-13",
    "end_date": "2017-04-13",
    "delegates": [
      {
        "class": "user",
        "url": "/results/p.php?i=2016PWFA01",
        "id": 6,
        "wca_id": "2016PWFA01",
        "name": "Trevor Bayer",
        "gender": "m",
        "country_iso2": "VE",
        "delegate_status": "delegate",
        "created_at": "2017-04-12T22:30:23.000Z",
        "updated_at": "2017-04-12T22:30:23.000Z",
        "teams": [

        ],
        "avatar": {
          "url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "thumb_url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "is_default": true
        },
        "email": "kenna.considine@howellstreich.name",
        "region": null,
        "senior_delegate_id": null
      }
    ],
    "organizers": [

    ]
  },
  {
    "class": "competition",
    "url": "/competitions/FooComp52015",
    "id": "FooComp52015",
    "name": "Foo Comp 5 2015",
    "website": "https://www.worldcubeassociation.org",
    "short_name": "Foo Comp 5 2015",
    "city": "San Francisco",
    "country_iso2": "US",
    "start_date": "2017-04-12",
    "end_date": "2017-04-12",
    "delegates": [
      {
        "class": "user",
        "url": "/results/p.php?i=2016STUX01",
        "id": 5,
        "wca_id": "2016STUX01",
        "name": "Mrs. Cole O'Conner",
        "gender": "m",
        "country_iso2": "DM",
        "delegate_status": "delegate",
        "created_at": "2017-04-12T22:30:23.000Z",
        "updated_at": "2017-04-12T22:30:23.000Z",
        "teams": [

        ],
        "avatar": {
          "url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "thumb_url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "is_default": true
        },
        "email": "brandt@keeling.io",
        "region": null,
        "senior_delegate_id": null
      }
    ],
    "organizers": [

    ]
  },
  {
    "class": "competition",
    "url": "/competitions/FooComp42015",
    "id": "FooComp42015",
    "name": "Foo Comp 4 2015",
    "website": "https://www.worldcubeassociation.org",
    "short_name": "Foo Comp 4 2015",
    "city": "San Francisco",
    "country_iso2": "US",
    "start_date": "2017-04-11",
    "end_date": "2017-04-11",
    "delegates": [
      {
        "class": "user",
        "url": "/results/p.php?i=2016GQIK01",
        "id": 4,
        "wca_id": "2016GQIK01",
        "name": "Lambert Kuhn",
        "gender": "m",
        "country_iso2": "KI",
        "delegate_status": "delegate",
        "created_at": "2017-04-12T22:30:22.000Z",
        "updated_at": "2017-04-12T22:30:22.000Z",
        "teams": [

        ],
        "avatar": {
          "url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "thumb_url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "is_default": true
        },
        "email": "kristoffer.collier@sporer.biz",
        "region": null,
        "senior_delegate_id": null
      }
    ],
    "organizers": [

    ]
  },
  {
    "class": "competition",
    "url": "/competitions/FooComp32015",
    "id": "FooComp32015",
    "name": "Foo Comp 3 2015",
    "website": "https://www.worldcubeassociation.org",
    "short_name": "Foo Comp 3 2015",
    "city": "San Francisco",
    "country_iso2": "US",
    "start_date": "2016-04-12",
    "end_date": "2016-04-12",
    "delegates": [
      {
        "class": "user",
        "url": "/results/p.php?i=2016KWEU01",
        "id": 3,
        "wca_id": "2016KWEU01",
        "name": "Tessie Ebert DVM",
        "gender": "m",
        "country_iso2": "BS",
        "delegate_status": "delegate",
        "created_at": "2017-04-12T22:30:22.000Z",
        "updated_at": "2017-04-12T22:30:22.000Z",
        "teams": [

        ],
        "avatar": {
          "url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "thumb_url": "/assets/missing_avatar_thumb-f0ea801c804765a22892b57636af829edbef25260a65d90aaffbd7873bde74fc.png",
          "is_default": true
        },
        "email": "aunta@parisian.org",
        "region": null,
        "senior_delegate_id": null
      }
    ],
    "organizers": [

    ]
  }
]
```

</details>

Whether or not we should be including all this in this particular API is a different question that I don't think we should address in this PR (this api already exists in production). I personally don't think this is worth changing until we document our api and do a `v1` of it. I think we're still in the "figuring out what we're doing" stage of things (my thoughts on this are kind of recorded [here](https://github.com/thewca/worldcubeassociation.org/pull/1348#issuecomment-291190459))